src/gurt/hash.c

```diff
@@ -13,6 +13,8 @@
 #include <gurt/list.h>
 #include <gurt/hash.h>
 
+#define HT_LOCKS_NR	64
```

Comment:
hmm, I think looks better still have one lock per bucket as your original patch, in dfuse/cart there are some hash table only with 8 or 16 buckets that is less than 64. 
your original patch looks fine, only concern is d_hhash_link_insert() looks can be refined a little bit as what i suggested before. But TBH personally I feel for the added HT_LOCKS_NR change I would rather keep your original patch.
It is just FYI, can see other person's thoughts, thanks.

Reply:
OK, i reverted this PR. as for your following suggestion, there are two places that try to lock all buckets.

d_hash_rec_insert_anonym:
  ch_key_init                            ------->just use atomic here looks not safe for me.
  ch_rec_insert_addref

and d_hhash_link_insert:
ch_rec_addref
hlink->hl_key = ptr_key

Anyway, let's try to address this in a following PR if necessary, thanks Xuezhao!

Reply:
> Anyway, let's try to address this in a following PR if necessary, thanks Xuezhao!

I pushed a small patch based on your current branch https://github.com/daos-stack/daos/pull/9954,
added a simple test tool for obj_open perf, testing on boro shows increased largely -
obj_open per second increased from 427.36 OPs/S to 313038.03 OPs/S.

please check and feel free to merge to your branch if fine with you, thanks.

Reply:
> > Anyway, let's try to address this in a following PR if necessary, thanks Xuezhao!
> 
> I pushed a small patch based on your current branch #9954, added a simple test tool for obj_open perf, testing on boro shows increased largely - obj_open per second increased from 427.36 OPs/S to 313038.03 OPs/S.

Nice.

> 
> please check and feel free to merge to your branch if fine with you, thanks.

Thanks, i merged it to this PR to make sure we don't introduce regression besides performance improvements.